### PR TITLE
Add Rover into the Kubernetes environment

### PR DESCRIPTION
**Pull request by mrproliu**
* Add Rover into the Kubernetes environment
* The Rover is not supported running in the docker-compose environment
* Add Trace and eBPF Profiling Cron Job

### CODE REVIEW

File: Makefile.in [hunk 1]
```diff
@@ -26,16 +26,17 @@ TAG ?= $(shell git rev-parse --short HEAD)
...
+SW_CLI_IMAGE ?= ghcr.io/apache/skywalking-cli/skywalking-cli:d2c95c3782f137518b09554331794c10e1457d95
 SW_EVENT_EXPORTER_IMAGE ?= ghcr.io/apache/skywalking-kubernetes-event-exporter/skywalking-kubernetes-event-exporter:8a012a3f968cb139f817189afb9b3748841bba22
 SW_AGENT_JAVA_IMAGE ?= ghcr.io/apache/skywalking-java/skywalking-java:8a0f79743cee6bc429218928f6afb296ed758ea4-java8
 
 SW_AGENT_NODEJS_BACKEND_VERSION ?= 59ef1aed6a404e2e8afffbb4b81ea849ae4f3026
 SW_AGENT_NODEJS_FRONTEND_VERSION ?= 1e31bd17dcebb616163d848fc435f3a2d4822fb8
 
-SW_SATELLITE_IMAGE ?= ghcr.io/apache/skywalking-satellite/skywalking-satellite:v0c52a26e37d778c0734ebcf9cd8a0e313c8ba682
+SW_SATELLITE_IMAGE ?= ghcr.io/apache/skywalking-satellite/skywalking-satellite:v1b2f7751b913f8baa58423bd146ff43758704ae7
+SW_ROVER_IMAGE ?= docker pull ghcr.io/apache/skywalking-rover/skywalking-rover:18e6e38d1b2a7cb8c2a933b3ac382bc6d7be7949
```
Comment (kezhenxu94) [inline Makefile.in:39]: This is obviously wrong

**Review by kezhenxu94 — CHANGES_REQUESTED** [review]

File: Makefile.in (hunk 1 above)
Comment (mrproliu) [reply Makefile.in]: Fixed.

File: deploy/platform/kubernetes/feature-rover/resources.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: rover-darmonset
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/feature-rover/resources.yaml:21]: Typo

File: deploy/platform/kubernetes/feature-rover/profiling.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          annotations:
+            sidecar.istio.io/inject: "false"
+        spec:
+          containers:
+            - name: ebpf-profiling-trigger
+              image: ghcr.io/apache/skywalking-cli/skywalking-cli:d2c95c3782f137518b09554331794c10e1457d95
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/feature-rover/profiling.yaml:36]: Replace with env var

File: deploy/platform/kubernetes/feature-rover/profiling.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              volumeMounts:
+              command:
+                - /bin/sh
+                - -c
+                - >
+                  /swctl --base-url=http://oap:12800/graphql profiling ebpf create fixed \
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/feature-rover/profiling.yaml:42]: The job is scheduled every 8 minutes but the duration is 9m. Would that be a problem?

**Comment by wu-sheng** [comment]
I think you missed the docs update new Features.

File: deploy/platform/kubernetes/feature-rover/profiling.yaml (hunk 4 above)
Comment (mrproliu) [reply deploy/platform/kubernetes/feature-rover/profiling.yaml:42]: Changed to the scheduled every 5 minutes in an hour.
For now, we only support the On_CPU profiling, so I just trying to profile when the envoy has requested.

File: docs/readme.md [hunk 5]
```diff
@@ -89,6 +89,7 @@ Currently, the features supported are:
...
 | `satellite`           | Deploy SkyWalking Satellite to load balance the monitoring data.                                                                           |                                                                                                                                       |
 | `function`            | Deploy [OpenFunction](https://openfunction.dev/) and export trace to SkyWalking.                                                           | Command [ofn](https://github.com/OpenFunction/cli) is required to run this feature.                                                   |
+| `rover`               | Deploy SkyWalking Rover and detect the processes in the Kubernetes environment.                                                            | Only Support Deploy to the Kubernetes environment, docker is not support.                                                             |
```
Comment (kezhenxu94) [inline docs/readme.md:92]:
```suggestion
| `rover`               | Deploy SkyWalking Rover and detect the processes in the Kubernetes environment.                                                            | Only support deployment in the Kubernetes environment, docker is not supported.                                                             |
```

File: deploy/platform/kubernetes/feature-rover/profiling.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+              image: ${SW_CLI_IMAGE}
+              volumeMounts:
+              command:
+                - /bin/sh
+                - -c
+                - >
+                  /swctl --base-url=http://oap:12800/graphql profiling ebpf create fixed \
+                    --service-name agentless::recommendation --labels mesh-envoy --duration=9m;
+                  /swctl --base-url=http://oap:12800/graphql profiling ebpf create fixed \
+                    --service-name agentless::app --labels mesh-envoy --duration=9m;
```
Comment (kezhenxu94) [inline deploy/platform/kubernetes/feature-rover/profiling.yaml:45]: Did you verify locally that all these services' base image has eBPF support? I'm not sure all these base images have the kernel that supports eBPF. Can you paste some screenshots?

**Comment by wu-sheng** [comment]
We don't need a high frequency scheduled task, is the task listed on the UI based on query duration or all available tasks listed there?

File: deploy/platform/kubernetes/feature-rover/profiling.yaml (hunk 6 above)
Comment (mrproliu) [reply deploy/platform/kubernetes/feature-rover/profiling.yaml:45]: All of these are profiling envoy applications. 
image

**Comment by mrproliu** [comment]
> We don't need a high frequency scheduled task.

Currently, Profiling tasks create a task every hour. Need to change to only creating one task a day?

> is the task listed on the UI based on query duration or all available tasks listed there?

List all available tasks.

**Comment by kezhenxu94** [comment]
> > We don't need a high frequency scheduled task.
> 
> 
> 
> Currently, Profiling tasks create one task every hour. Need to change to only creating one task every day?
> 
> 
> 
> > is the task listed on the UI based on query duration or all available tasks listed there?
> 
> 
> 
> List all available profiling tasks. but profiling schedule only queries with duration.

If the ui lists all tasks, we can keep the schedule rate as currently set (every hour). The reason here is that we might restart/redeploy the env and we can have the data soon even after redeployment. This patch looks good to me now (merge if you are all set)

**Review by kezhenxu94 — APPROVED** [review]

**Comment by wu-sheng** [comment]
> If the ui lists all tasks, we can keep the schedule rate as currently set (every hour).

Could we set an initial task(waiting for OAP to be ready) and a per-day schedule to ease the ElasticSearch payload?

**Comment by mrproliu** [comment]
Currently, the eBPF Profiling uses `ON_CPU` on the envoy applications. It's not easy to capture the stacks, that's why I  extended the duration. I will change the trace and eBPF profiling to a daily job.

**Comment by wu-sheng** [comment]
Pending on https://github.com/apache/skywalking/pull/9034 and @Fine0830 's next PR to remove `schedule duration condition`, then this could be merged and hosted on demo.skywalking.a.o

**Comment by Fine0830** [comment]
> Pending on [apache/skywalking#9034](https://github.com/apache/skywalking/pull/9034) and @Fine0830 's next PR to remove `schedule duration condition`, then this could be merged and hosted on demo.skywalking.a.o

I have removed the `duration` filed from the schedule condition. It is in https://github.com/apache/skywalking-booster-ui/pull/83 .

**Comment by wu-sheng** [comment]
@mrproliu Do another UI submodule update?

**Comment by mrproliu** [comment]
After [PR](https://github.com/apache/skywalking/pull/9035), I will sync the image.

**Pull request merged**